war/src/test/java/io/lumeer/remote/rest/permissions/OrganizationServicePermissionsIntegrationTest.java

```diff
@@ -0,0 +1,404 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) since 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.remote.rest.permissions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.lumeer.api.dto.JsonOrganization;
+import io.lumeer.api.dto.JsonPermission;
+import io.lumeer.api.dto.JsonProject;
+import io.lumeer.api.model.Organization;
+import io.lumeer.api.model.Permission;
+import io.lumeer.api.model.Project;
+import io.lumeer.api.model.Role;
+import io.lumeer.core.AuthenticatedUser;
+import io.lumeer.core.facade.OrganizationFacade;
+import io.lumeer.core.facade.ProjectFacade;
+import io.lumeer.engine.IntegrationTestBase;
+import io.lumeer.storage.api.dao.OrganizationDao;
+import io.lumeer.storage.api.dao.ProjectDao;
+import io.lumeer.test.util.LumeerAssertions;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.*;
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@RunWith(Arquillian.class)
+public class OrganizationServicePermissionsIntegrationTest extends IntegrationTestBase {
```

Comment:
You should inherit from the new `ServiceIntegrationTestBase` class so you don't have to create a new HTTP client in each test and can use `client` attribute from the parent class instead. It will also handle closing the client after the tests so there will be no hanging connections if a test fails.